static_src/components/activity_log_item.jsx

```diff
@@ -0,0 +1,135 @@
+
+import moment from 'moment';
+import style from 'cloudgov-style/css/cloudgov-style.css';
+import React from 'react';
+
+import createStyler from '../util/create_styler';
+import RouteStore from '../stores/route_store';
+import ServiceInstanceStore from '../stores/service_instance_store';
+
+export default class ActivityLogItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showRawJson: false,
+      relativeTimestamp: true
+    };
+
+    this.styler = createStyler(style);
+    this._onChange = this._onChange.bind(this);
+    this.toggleRawJson = this.toggleRawJson.bind(this);
+    this.toggleTimestampType = this.toggleTimestampType.bind(this);
+  }
+
+  componentDidMount() {
+    RouteStore.addChangeListener(this._onChange);
+  }
+
+  componentWillUnmount() {
+    RouteStore.removeChangeListener(this._onChange);
+  }
+
+  _onChange() {
+    this.setState({});
+  }
+
+  formatTimestamp(timestamp) {
+    if (this.state.relativeTimestamp) return moment(timestamp).fromNow();
+
+    return moment(timestamp).format('MMMM Do YYYY, h:mm:ss a');
+  }
+
+  toggleRawJson() {
+    const current = this.state.showRawJson;
+    this.setState({ showRawJson: !current });
+  }
+
+  toggleTimestampType() {
+    this.setState({ relativeTimestamp: !this.state.relativeTimestamp });
+  }
+
+  get code() {
+    let content;
+    if (this.state.showRawJson) {
+      content = (
+        <div className={ this.styler('activity-log-item-raw') }>
+          <code>
+            <pre>
+              { JSON.stringify(this.props.item, null, 2) }
+            </pre>
+          </code>
+        </div>
+      );
+    }
+
+    return content;
+  }
+
+  get content() {
+    let content = `${this.props.item.type} isn't handled`;
+    const item = this.props.item;
+    const metadata = item.metadata;
+    const route = RouteStore.get(metadata.route_guid);
+    const url = (route) ? `${route.host}${route.domain}/${route.path}` : 'a url';
+    const link = (route) ? (<a href={ `//${url}` }>{ url }</a>) : url;
+
+    // TODO: if route is not found, trigger fetch action to get it
```

Comment:
I actually don't think actions should be called in JSX components because they lead to two dispatches happening at same time which leads to error.